src/anomalib/utils/post_processing.py

```diff
@@ -117,6 +117,8 @@ def superimpose_anomaly_map(
         np.ndarray: Image with anomaly map superimposed on top of it.
     """
     anomaly_map = anomaly_map_to_color_map(anomaly_map.squeeze(), normalize=normalize)
+    height, width = anomaly_map.shape[:2]
+    image = cv2.resize(image, (width, height))
```

Comment:
can you explain why is this needed?

Reply:
This is to ensure that the image and the anomaly map have the same shape. During visualization we load the original image from the filesystem, so it may not have the same dimensions as the anomaly map predictions. This change ensures that both have the same shape before superimposing the heatmap on the image.